include/autoppl/expr_builder.hpp

```diff
@@ -156,6 +157,22 @@ inline constexpr auto bernoulli(ProbType&& p_expr)
     return expr::Bernoulli(wrap_p_expr);
 }
 
+/*
+ * Builds a Discrete expression only when the parameter
+ * is a valid discrete distribution parameter type.
+ * See var_expr.hpp for more information.
+ */
+template <class ProbType
+        , class = std::enable_if_t<
+            details::is_valid_dist_param_v<ProbType>
+        > >
+inline constexpr auto discrete(ProbType&& p_expr)
+{
+    using p_t = details::convert_to_param_t<ProbType>;
```

Comment:
This converts value types, it doesn't convert containers.

Reply:
Yup do we need support for containers? I can write one if needed.